pkg/engine/globalcontext/controller.go

```
@@ -0,0 +1,189 @@
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const (
+	resyncPeriod = 15 * time.Second
+	maxRetries   = 3
+)
+
+type globalContextController struct {
+	logger        logr.Logger
+	kyvernoClient versioned.Interface
+	dynamicClient dynamic.Interface
+	store         *store.Store
+
+	queue           workqueue.RateLimitingInterface
+	gctxentInformer kyvernov2alpha1informers.GlobalContextEntryInformer
+	gctxentLister   kyvernov2alpha1listers.GlobalContextEntryLister
+	informerSynced  cache.InformerSynced
+}
+
+func NewController(logger logr.Logger, kyvernoClient versioned.Interface, dclient dynamic.Interface, storage *store.Store) controllers.Controller {
+	kyvernoinfoermer := kyvernoinfoermers.NewSharedInformerFactory(kyvernoClient, resyncPeriod)
+	gctxentInformer := kyvernoinfoermer.Kyverno().V2alpha1().GlobalContextEntries()
+	gctxentLister := gctxentInformer.Lister()
+	informerSynced := gctxentInformer.Informer().HasSynced
+
+	return &globalContextController{
+		logger:          logger,
+		kyvernoClient:   kyvernoClient,
+		dynamicClient:   dclient,
+		store:           storage,
+		queue:           workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "globalcontext"),
+		gctxentInformer: gctxentInformer,
+		gctxentLister:   gctxentLister,
+		informerSynced:  informerSynced,
+	}
+}
+
+func (gc *globalContextController) Run(ctx context.Context, workers int) {
+	logger := gc.logger
+
+	defer utilruntime.HandleCrash()
+	defer gc.queue.ShutDown()
+
+	logger.Info("starting")
+	defer logger.Info("shutting down")
+
+	if !cache.WaitForNamedCacheSync("GlobalContextController", ctx.Done(), gc.informerSynced) {
+		return
+	}
+
+	_, _ = gc.gctxentInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Is it okay to ignore the error and hasSynced function here?